diff --git a/CMakeLists.txt b/CMakeLists.txt
index e49548fb..0bd8b8ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,7 +22,7 @@ if(NOT DEFINED QT_VERSION_MAJOR)
     find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core Network Xml)
 endif()
 message("-- Choosing Qt ${QT_VERSION_MAJOR}")
-find_package(Qt${QT_VERSION_MAJOR} 5.9.2 REQUIRED COMPONENTS Core Network Xml)
+find_package(Qt${QT_VERSION_MAJOR} 5.6 REQUIRED COMPONENTS Core Network Xml)
 
 # QCA (optional)
 find_package(Qca-qt${QT_VERSION_MAJOR} QUIET)
